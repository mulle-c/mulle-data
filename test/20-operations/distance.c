#include <mulle-data/mulle-data.h>

#include <stdio.h>

static void  mulle_range_print( struct mulle_range a)
{
   if( ! a.length)
      printf( "NULL");  
   else
      printf( "%td-%td (%td)", a.location, a.location + a.length - 1, a.length);  
}


int   main( int argc, char *argv[])
{
   struct mulle_range   a;
   uintptr_t            c;

   for( a.location = 0; a.location < 100; a.location += 25)
      for( a.length = 0; a.length < 100; a.length += 25)
         for( c = 0; c <= 150; c += 25)
         {
            mulle_range_print( a);
            printf( " distance to %td = %td\n",
                  c, mulle_range_distance_to_location( a, c));
         }

   return( 0);
}


/*
 * extension : mulle-sde/c-test-library-demo
 * directory : demo/all
 * template  : .../hello.c
 * Suppress this comment with `export MULLE_SDE_GENERATE_FILE_COMMENTS=NO`
 */
