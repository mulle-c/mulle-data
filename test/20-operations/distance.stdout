NULL distance to 0 = -1
NULL distance to 25 = -1
NULL distance to 50 = -1
NULL distance to 75 = -1
NULL distance to 100 = -1
NULL distance to 125 = -1
NULL distance to 150 = -1
0-24 (25) distance to 0 = 0
0-24 (25) distance to 25 = 1
0-24 (25) distance to 50 = 26
0-24 (25) distance to 75 = 51
0-24 (25) distance to 100 = 76
0-24 (25) distance to 125 = 101
0-24 (25) distance to 150 = 126
0-49 (50) distance to 0 = 0
0-49 (50) distance to 25 = 0
0-49 (50) distance to 50 = 1
0-49 (50) distance to 75 = 26
0-49 (50) distance to 100 = 51
0-49 (50) distance to 125 = 76
0-49 (50) distance to 150 = 101
0-74 (75) distance to 0 = 0
0-74 (75) distance to 25 = 0
0-74 (75) distance to 50 = 0
0-74 (75) distance to 75 = 1
0-74 (75) distance to 100 = 26
0-74 (75) distance to 125 = 51
0-74 (75) distance to 150 = 76
NULL distance to 0 = -1
NULL distance to 25 = -1
NULL distance to 50 = -1
NULL distance to 75 = -1
NULL distance to 100 = -1
NULL distance to 125 = -1
NULL distance to 150 = -1
25-49 (25) distance to 0 = 25
25-49 (25) distance to 25 = 0
25-49 (25) distance to 50 = 1
25-49 (25) distance to 75 = 26
25-49 (25) distance to 100 = 51
25-49 (25) distance to 125 = 76
25-49 (25) distance to 150 = 101
25-74 (50) distance to 0 = 25
25-74 (50) distance to 25 = 0
25-74 (50) distance to 50 = 0
25-74 (50) distance to 75 = 1
25-74 (50) distance to 100 = 26
25-74 (50) distance to 125 = 51
25-74 (50) distance to 150 = 76
25-99 (75) distance to 0 = 25
25-99 (75) distance to 25 = 0
25-99 (75) distance to 50 = 0
25-99 (75) distance to 75 = 0
25-99 (75) distance to 100 = 1
25-99 (75) distance to 125 = 26
25-99 (75) distance to 150 = 51
NULL distance to 0 = -1
NULL distance to 25 = -1
NULL distance to 50 = -1
NULL distance to 75 = -1
NULL distance to 100 = -1
NULL distance to 125 = -1
NULL distance to 150 = -1
50-74 (25) distance to 0 = 50
50-74 (25) distance to 25 = 25
50-74 (25) distance to 50 = 0
50-74 (25) distance to 75 = 1
50-74 (25) distance to 100 = 26
50-74 (25) distance to 125 = 51
50-74 (25) distance to 150 = 76
50-99 (50) distance to 0 = 50
50-99 (50) distance to 25 = 25
50-99 (50) distance to 50 = 0
50-99 (50) distance to 75 = 0
50-99 (50) distance to 100 = 1
50-99 (50) distance to 125 = 26
50-99 (50) distance to 150 = 51
50-124 (75) distance to 0 = 50
50-124 (75) distance to 25 = 25
50-124 (75) distance to 50 = 0
50-124 (75) distance to 75 = 0
50-124 (75) distance to 100 = 0
50-124 (75) distance to 125 = 1
50-124 (75) distance to 150 = 26
NULL distance to 0 = -1
NULL distance to 25 = -1
NULL distance to 50 = -1
NULL distance to 75 = -1
NULL distance to 100 = -1
NULL distance to 125 = -1
NULL distance to 150 = -1
75-99 (25) distance to 0 = 75
75-99 (25) distance to 25 = 50
75-99 (25) distance to 50 = 25
75-99 (25) distance to 75 = 0
75-99 (25) distance to 100 = 1
75-99 (25) distance to 125 = 26
75-99 (25) distance to 150 = 51
75-124 (50) distance to 0 = 75
75-124 (50) distance to 25 = 50
75-124 (50) distance to 50 = 25
75-124 (50) distance to 75 = 0
75-124 (50) distance to 100 = 0
75-124 (50) distance to 125 = 1
75-124 (50) distance to 150 = 26
75-149 (75) distance to 0 = 75
75-149 (75) distance to 25 = 50
75-149 (75) distance to 50 = 25
75-149 (75) distance to 75 = 0
75-149 (75) distance to 100 = 0
75-149 (75) distance to 125 = 0
75-149 (75) distance to 150 = 1
